features/dd-sdk-android-flags/src/test/kotlin/com/datadog/android/flags/featureflags/FlagsClientTest.kt

```diff
@@ -39,252 +35,202 @@ internal class FlagsClientTest {
     lateinit var mockInternalLogger: InternalLogger
 
     @Mock
-    lateinit var mockFlagsProvider: FlagsProvider
+    lateinit var mockFlagsClient: FlagsClient
 
     @Mock
     lateinit var mockSecondSdkCore: FeatureSdkCore
 
     @Mock
-    lateinit var mockSecondFlagsProvider: FlagsProvider
-
-    private lateinit var fakeSdkCoreName: String
+    lateinit var mockSecondFlagsClient: FlagsClient
 
     @BeforeEach
-    fun `set up`(forge: Forge) {
-        fakeSdkCoreName = forge.anAlphabeticalString()
-        whenever(mockSdkCore.internalLogger) doReturn mockInternalLogger
-        whenever(mockSdkCore.name) doReturn fakeSdkCoreName
-        whenever(mockSecondSdkCore.internalLogger) doReturn mockInternalLogger
-
-        // Clean state before each test
-        FlagsClient.clear()
+    fun `set up`() {
+        whenever(mockSdkCore.internalLogger).thenReturn(mockInternalLogger)
+        whenever(mockSecondSdkCore.internalLogger).thenReturn(mockInternalLogger)
+        whenever(mockSdkCore.name).thenReturn("test-sdk")
+        whenever(mockSecondSdkCore.name).thenReturn("test-sdk-2")
     }
 
     @AfterEach
     fun `tear down`() {
         FlagsClient.clear()
     }
 
-    // region isRegistered()
+    // region Static Flag Resolution Methods
 
     @Test
-    fun `M return false W isRegistered() {no provider registered}`() {
+    fun `M return no-op default W instance() + resolveBooleanValue() {no client registered for SDK core}`() {
```

Comment:
```suggestion
    fun `M return no-op default W instance() + resolveBooleanValue() {no client registered for SDK core}`(
      @StringForgery fakeFlagName: String,
      @BooleanForgery fakeFlagDefaultValue: Boolean
    ) {
```

we don't usually use defined values in tests, but random ones